frontend/libs/studio-content-library/src/ContentLibrary/LibraryBody/LibraryBody.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import classes from './LibraryBody.module.css';
+import { PagesRouter } from './PagesRouter';
+import { InfoBox } from './InfoBox';
+import type { PagePropsMap, PagesConfig } from '../../types/PagesProps';
+import type { PageName } from '../../types/PageName';
+import type { PageComponent } from '../../utils/router/RouterRouteMapper';
+
+type LibraryBodyProps<T extends PageName = 'landingPage'> = {
+  Component: PageComponent<Required<PagePropsMap>[T]>;
+  pages: PagesConfig;
+  currentPage: PageName;
+};
+
+export function LibraryBody<T extends PageName = 'landingPage'>({
+  Component,
+  pages,
+  currentPage,
+}: LibraryBodyProps) {
+  const componentProps: Required<PagePropsMap>[T] = pages[currentPage]
+    ?.props as Required<PagePropsMap>[T];
```

Comment:
Så vidt jeg kan se, brukes alltid `PagePropsMap` på denne måten, med en henvisning til `[T]`. Da er det bedre å definere `PagePropsMap` som en generisk type, slik at vi kan skrive `PagePropsMap<T>` i stedet for `PagePropsMap[T]`. Da trenger vi kanskje heller ikke å caste typen her.

Slik kan vi definere `PagePropsMap` som en generisk type:
```
type PagePropsMap<P extends PageName> = {
  landingPage: {};
  codeList: CodeListProps;
  images: ImagesProps;
}[P];
```

Reply:
Endret typen, men virker som jeg fortsatt må caste 🫣

Reply:
Ah, det kan være på grunn av spørsmålstegnet foran `.props`. Jeg la ikke merke til det. Trenger vi det? Vet vi ikke at at `pages[currentPage]` eksisterer her?

Reply:
Er ikke spm-tegn der lenger, men er sånn:

```
const componentPropsAreExternal = currentPage !== 'landingPage';

  const componentProps: Required<PagePropsMap>[T] =
    componentPropsAreExternal && (pages[currentPage].props as Required<PagePropsMap>[T]);
 ```
 
 fordi landingPage ikke har external props fordi den ikke er ekstern i det hele tatt

Reply:
Ah, jeg har sett på utdatert kode. Men trenger vi denne sjekken da? Vil ikke `pages['landingPage']` bare bli et tomt objekt? Nå som du har fjernet spørsmålstegnene fra `PagePropsMap`, trenger du for øvrig ikke `Required` her.

Reply:
Nvm, jeg som så på feil kode. Tror du har rett! La oss se på det i en huddle 😄